packages/compass-assistant/src/assistant-global-state.tsx

```diff
@@ -0,0 +1,62 @@
+import type { ConnectionInfo } from '@mongodb-js/connection-info';
+import type {
+  WorkspaceTab,
+  CollectionTabInfo,
+} from '@mongodb-js/workspace-info';
+import React, { useEffect } from 'react';
+
+export type GlobalState = {
+  currentActiveConnections: ConnectionInfo[];
+  currentWorkspace: WorkspaceTab | null;
+  currentWorkspaceCollectionInfo: CollectionTabInfo | null;
+  currentQuery: object | null;
+  currentAggregation: object | null;
+};
+
+const INITIAL_STATE: GlobalState = {
+  currentActiveConnections: [],
+  currentWorkspace: null,
+  currentWorkspaceCollectionInfo: null,
+  currentQuery: null,
+  currentAggregation: null,
+};
+
+const AssistantGlobalStateContext = React.createContext<GlobalState>({
+  ...INITIAL_STATE,
+});
+
+const AssistantGlobalSetStateContext = React.createContext<
+  React.Dispatch<React.SetStateAction<GlobalState>>
+>(() => undefined);
+
+export const AssistantGlobalStateProvider: React.FunctionComponent = ({
+  children,
+}) => {
+  const [globalState, setGlobalState] = React.useState({ ...INITIAL_STATE });
+  return (
+    <AssistantGlobalStateContext.Provider value={globalState}>
+      <AssistantGlobalSetStateContext.Provider value={setGlobalState}>
+        {children}
+      </AssistantGlobalSetStateContext.Provider>
+    </AssistantGlobalStateContext.Provider>
+  );
+};
+
+export function useSyncAssistantGlobalState<T extends keyof GlobalState>(
```

Comment:
Not sure how it should look like exactly, but you might want some sort of cleanup logic when the hook gets "unmounted" (so maybe not part of the same effect that updates the state, but an extra effect that sets stateKey back to null in cleanup?)

Reply:
Done in https://github.com/mongodb-js/compass/pull/7629/commits/851cb889a3dcacb9c789a478fae7a20c239f702d